components/scream/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -0,0 +1,965 @@
+          const int lid = lids_pids(i,0);
+          const int pid = lids_pids(i,1);
+          const int lidpos = i - pid_lid_start(pid);
+          const int offset = f_pid_offsets(pid);
+
+          buf (offset + lidpos) = v(lid);
+        });
+                             
+      } break;
+      case LayoutType::Vector2D:
+      {
+        auto v = f.get_view<const Real**>();
+        const int ndims = fl.dim(1);
+        auto policy = ESU::get_default_team_policy(num_send_gids,ndims);
+        Kokkos::parallel_for(policy,
+                             KOKKOS_LAMBDA(const MemberType& team){
+          const int i = team.league_rank();
+          const int lid = lids_pids(i,0);
+          const int pid = lids_pids(i,1);
+          const int lidpos = i - pid_lid_start(pid);
+          const int offset = f_pid_offsets(pid);
+
+          Kokkos::parallel_for(Kokkos::TeamThreadRange(team,ndims),
+                               [&](const int idim) {
+            buf(offset + lidpos*ndims + idim) = v(lid,idim);
+          });
+        });
+      } break;
+      case LayoutType::Scalar3D:
+      {
+        auto v = f.get_view<const Real**>();
+        const int nlevs = fl.dims().back();
+        auto policy = ESU::get_default_team_policy(num_send_gids,nlevs);
+        Kokkos::parallel_for(policy,
+                             KOKKOS_LAMBDA(const MemberType& team){
+          const int i = team.league_rank();
+          const int lid = lids_pids(i,0);
+          const int pid = lids_pids(i,1);
+          const int lidpos = i - pid_lid_start(pid);
+          const int offset = f_pid_offsets(pid);
+
+          Kokkos::parallel_for(Kokkos::TeamThreadRange(team,nlevs),
+                               [&](const int ilev) {
+            buf(offset + lidpos*nlevs + ilev) = v(lid,ilev);
+          });
+        });
+      } break;
+      case LayoutType::Vector3D:
+      {
+        auto v = f.get_view<const Real***>();
+        const int ndims = fl.dim(1);
+        const int nlevs = fl.dims().back();
+        auto policy = ESU::get_default_team_policy(num_send_gids,ndims*nlevs);
+        Kokkos::parallel_for(policy,
+                             KOKKOS_LAMBDA(const MemberType& team){
+          const int i = team.league_rank();
+          const int lid = lids_pids(i,0);
+          const int pid = lids_pids(i,1);
+          const int lidpos = i - pid_lid_start(pid);
+          const int offset = f_pid_offsets(pid);
+
+          Kokkos::parallel_for(Kokkos::TeamThreadRange(team,ndims*nlevs),
+                               [&](const int idx) {
+            const int idim = idx / nlevs;
+            const int ilev = idx % nlevs;
+            buf(offset + lidpos*ndims*nlevs + idim*nlevs + ilev) = v(lid,idim,ilev);
+          });
+        });
+      } break;
+
+      default:
+        EKAT_ERROR_MSG ("Unexpected field rank in CoarseningRemapper::pack.\n"
+            "  - MPI rank  : " + std::to_string(m_comm.rank()) + "\n"
```

Comment:
Two different kinds of ranks!

Reply:
I know, I was struggling to make the message clear, without any confusion.